File: Leetcode/1290/1290.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        auto cur = head;
        string s;
        while (cur) {
            if (cur->val == 1) {
                s.push_back('1');
            } else {
                s.push_back('0');
            }
            cur = cur->next;
        }
        return stoll(s, nullptr, 2);
    }
};

File: Leetcode/1290/1290.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func getDecimalValue(head *ListNode) int {
	node := head
	var runes []rune
	for node != nil {
		runes = append(runes, rune(node.Val+'0'))
		node = node.Next
	}
	ans, err := strconv.ParseInt(string(runes), 2, 64)
	if err != nil {
		log.Fatalln(err)
	}
	return int(ans)
}